src/utils/Processing.ts

```diff
@@ -99,21 +102,82 @@ export function loadExternalDependencies(
   return dependencyDefs;
 }
 
-function getLakeOfFHIR(inDir: string): LakeOfFHIR {
-  const files = getFilesRecursive(inDir).filter(file => file.endsWith('.json'));
-  logger.info(`Found ${files.length} JSON files.`);
+export function getLakeOfFHIR(inDir: string, fileType: string): LakeOfFHIR {
+  const files = getFilesRecursive(inDir);
+  const jsonFiles = files.filter(f => f.endsWith('.json'));
+  const xmlFiles = files.filter(f => f.endsWith('.xml'));
   const docs: WildFHIR[] = [];
+
+  if (fileType === 'json-only') {
+    logger.info(`Found ${jsonFiles.length} JSON files.`);
+    loadPrimaryFiles(jsonFiles, docs);
+    const nonDuplicateXMLFile = findNonDuplicateSecondaryFile(xmlFiles, docs);
+    if (nonDuplicateXMLFile) {
+      // We only find the first non-duplicate file and warn on it, because warnings for every non-duplicate may be annoying
+      logger.warn(
+        `The XML definition at ${nonDuplicateXMLFile} will be ignored since GoFSH is running in "json-only" mode.` +
+          ' To process XML definitions along with JSON, set the "-t" flag to "json-and-xml".' +
+          ' To process only XML definitions, set the "-t" flag to "xml-only".'
+      );
```

Comment:
I feel like only printing one could be misleading (it makes it look like there is only one).  Can we get a count and indicate that?
> 17 XML definitions found without corresponding JSON definitions (for example, ${nonDuplicateXMLFile}). These definitions will be ignored since GoFSH is running in "json-only" mode....